.github/workflows/pytest-gpu.yml

```diff
@@ -0,0 +1,64 @@
+# Adapted from https://github.com/weeyin83/vm-actions
+name: Deploy VM on Azure and run GPU tests
+
+env:
+  OUTPUT_PATH: ${{ github.workspace }}
+
+on: [push]
+
+jobs:
+
+  # Deploy VM in Azure
+  DeployVM:
+    runs-on: ubuntu-latest
+
+    steps:
+    # checkout code from repo
+    - name: checkout repo
+      uses: actions/checkout@v1
+
+    - name: extract branch name
+      shell: bash
+      run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+      id: extract_branch
+
+    - name: look for ps1 file
+      run: |
+        ls '${{ env.OUTPUT_PATH }}/PSA'
+    - name: deploy VM
+      env:
+        RESOURCE_GROUP: ${{ secrets.RESOURCE_GROUP }}
+        RESOURCE_GROUP_REGION: uksouth
+        SERVER_NAME: gpu_ci_vm
+      run: >
+        pwsh -command "& '${{ env.OUTPUT_PATH }}\PSA\deployVM.ps1'"
+        -servicePrincipal ${{ secrets.SERVICE_PRINCIPAL_APPID }}
+        -servicePrincipalSecret ${{ secrets.SERVICE_PRINCIPAL_SECRET }}
+        -servicePrincipalTenantId ${{ secrets.SERVICE_PRINCIPAL_TENANTID }}
+        -azureSubscriptionName ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        -resourceGroupName $RESOURCE_GROUP
+        -resourceGroupNameRegion $RESOURCE_GROUP_REGION
+        -serverName $SERVER_NAME
+        -adminLogin ${{ secrets.ADMIN_LOGIN }}
+        -adminPassword ${{ secrets.ADMIN_PASSWORD }}
+
+    - name: set host
+      run: echo ::set-output name=action_host::$(az vm show -d -g ${{ secrets.RESOURCE_GROUP }} -n gpu_ci_vm --query publicIps -o tsv)
+      id: host
+
+    - name: executing remote ssh commands test
+      uses: fifsky/ssh-action@master
+      with:
+        command: |
+          export PATH=~/programs/install/bin:$PATH
```

Comment:
are the PATH and LD_LIBRARY_PATH exports so that `clang` and the cuda libs are visible ? if so, could we add a comment? 'cause I'm sure I will forget one day

Reply:
Yes. Added a comment.